modules/shortcodes/email-subscribe.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/*
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ */
+class Jetpack_Email_Subscribe {
+
+	static $shortcode = 'jetpack-email-subscribe';
+
+	static $css_classname_prefix = 'jetpack-email-subscribe';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		// Do not load this at all if it's neither a WPCOM or a connected JP site.
+		if ( ! ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || Jetpack::is_active() ) ) {
+			return null;
+		}
+
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts_and_styles() {
+		wp_register_script( 'jetpack-email-subscribe', Jetpack::get_file_url_for_environment( '_inc/build/shortcodes/js/jetpack-email-subscribe.min.js', 'modules/shortcodes/js/jetpack-email-subscribe.js' ), array( 'jquery' ) );
+		wp_register_style( 'jetpack-email-subscribe', plugins_url( '/css/jetpack-email-subscribe.css', __FILE__ ) );
+	}
+
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		$this->register_scripts_and_styles();
+		$this->register_shortcode();
+	}
+
+	function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
+	}
+
+
+	function parse_shortcode( $attrs, $content = false ) {
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'blog_id'     => $this->get_blog_id(),
+			'title'       => __( 'Join my email list', 'jetpack' ),
+			'email_placeholder' => __( 'Enter your email', 'jetpack' ),
+			'submit_label' => __( 'Join My Email List', 'jetpack' ),
+			'consent_text' => __( 'By clicking submit, you agree to share your email address with the site owner and MailChimp to receive marketing, updates, and other emails from the site owner. Use the unsubscribe link in those emails to opt out at any time.', 'jetpack' ),
+			'processing_label' => __( 'Processing...', 'jetpack' ),
+			'success_label' => __( 'Success! You\'ve been added to the list.', 'jetpack' ),
+			'error_label' => __( "Oh no! Unfortunately there was an error.\nPlease try reloading this page and adding your email once more.", 'jetpack' ),
+			'classname' => self::$css_classname_prefix,
+			'dom_id' => uniqid( self::$css_classname_prefix . '_', false ),
+		), $attrs );
+
+
+		if ( ! wp_script_is( 'jetpack-email-subscribe', 'enqueued' ) ) {
+			wp_enqueue_script( 'jetpack-email-subscribe' );
+		}
+
+		if( ! wp_style_is( 'jetpack-email-subscribe', 'enqueue' ) ) {
```

Comment:
You have a missing space here. Could you try to enable linting in your editor? That should help you catch the different coding standards issues (indenting, spaces) in that file.

Reply:
🤦‍♂️  OMG, funny story:
Since I work on payments stuff, I had to configure special install of PHPCS because payments standards are much more strict.
But PHPStorm can hold only 1 configuration in settings and the one from payments dies silently when used in other repos.
Which reports 0 errors to PHPStorm, so it looks like the file is clear.

You are right. That file was a mess. Fixed now.